Add Replicate demo

ArielReplicate opened the pull request:
Hey @GuyTevet ! 👋 

Thank for this cool project.

This pull request makes it possible to run your model inside a Docker environment, which makes it easier for other people to run it.  We're using an open source tool called [Cog](https://github.com/replicate/cog) to make this process easier.

This also means we can make a web page where other people can run your model! 
View it here: https://replicate.com/arielreplicate/motion_diffusion_model

Replicate also have an API, so people can easily run your model from their code:

```python
import replicate
model = replicate.models.get("[arielreplicate/motion_diffusion_model]")
model.predict(...)
```

If you'd like to modify the Replicate page, let me know and I can transfer ownership to your account.

In case you're wondering who I am, I'm from [Replicate](https://replicate.com/home), where we're trying to make machine learning reproducible. We got frustrated that we couldn't run all the really interesting ML work being done. So, we're going round implementing models we like. 😊

GuyTevet commented:
This is so cool! Thank you Ariel!
Are those files needed for the Replicate demo and API? If so, how did you manage to put the demo on air without them being pushed to the repo?

Let's accelerate the demo by putting all the repetitions in the same batch. Currently, each is running in a different batch (bs=1) with our non-optimal text_prompt implementation.

GuyTevet commented on predict.py:
```diff
@@ -0,0 +1,221 @@
+import subprocess
```
Can you move this file to ./sample/ ?

ArielReplicate commented:
Thanks @GuyTevet, I'm happy you liked the demo!
The files aren't needed for the demo to run online but are needed to push changes into it using cog.
For example if you want to add control over more inference parameters like the "guidance_param" you can edit _predict.py_ and update the demo.

I changed the code to run with batch_size=num_repetitions.
I also moved predict.py into samples/. and cleaned the code from code repetitions

Please tell me if you think we can further improve the demo or if you want to put it under your Replicate user so you can edit yourself.

GuyTevet commented:
Looks good! Thanks again! Merging
